tests/test_mesh_creation_functions.py

```diff
@@ -211,6 +211,46 @@ def test_mesh_creation_functions_arc_segment_2d(
     assert_results_equal(get_corresponding_reference_file_path(), input_file)
 
 
+def test_mesh_creation_functions_arc_segment_2d_prescribed_node_positions_of_elements(
+    assert_results_equal, get_corresponding_reference_file_path
+):
+    """Create circular segments in 2D with prescribed node position."""
+
+    # Create input file.
+    input_file = InputFile()
+
+    # Add material and function.
+    mat = MaterialReissner(radius=0.1)
+
+    # Create a arc segment similar to the equally spaced one,
+    # but based on the provided node_positions_of_elements.
+    create_beam_mesh_arc_segment_2d(
+        input_file,
+        Beam3rHerm2Line3,
+        mat,
+        [1.0, 2.0, 0.0],
+        1.5,
+        np.pi * 0.25,
+        np.pi * (1.0 + 1.0 / 3.0),
+        node_positions_of_elements=[0.0, 0.2, 0.4, 0.6, 0.8, 1.0],
+    )
+
+    # Create a 2d segments with different element sizes
+    create_beam_mesh_arc_segment_2d(
+        input_file,
+        Beam3rHerm2Line3,
+        mat,
+        [1.0, 2.0, 0.0],
+        2.0,
+        np.pi * 0.25,
+        np.pi * (1.0 + 1.0 / 3.0),
+        node_positions_of_elements=[0, 0.5, 0.75, 0.875, 1.0],
```

Comment:
```suggestion
        node_positions_of_elements=[0, 1.0/ 3.0, 1.0],
```
Should be sufficient.